src/benchmarks/micro/corefx/System.Buffers/BinaryReadAndWriteTests.cs

```diff
@@ -0,0 +1,300 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using System.Net;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using MicroBenchmarks;
+
+using static System.Buffers.Binary.BinaryPrimitives;
+
+namespace System.Buffers.Binary.Tests
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.CoreCLR)]
+    public class BinaryReadAndWriteTests
+    {
+        private readonly static byte[] _arrayLE = GetSpanLE().ToArray();
+        private readonly static byte[] _arrayBE = GetSpanBE().ToArray();
+        private readonly static int[] _oneThousandIntegers = new int[1000];
```

Comment:
How are you deciding whether to have numeric inputs like this in a Params and the various fields/properties initialized in a global setup routine, versus having static members initialized directly?  I see both approaches in this PR, and I'm not clear on why some are done one way and some are done another.

Reply:
If all the benchmarks are using the same input data of the same size and it should not change in the future, I initialize everything in ctor. 

Otherwise, I use Params + Setups.

Reply:
Thanks.  What's the benefit of that approach versus putting tests into their own class?

Reply:
Setup vs ctor:

- Setup can use values from `[Params]`, ctor can not (they are set after we run the ctor)
- Setup can be async, ctor can not
- ctor can initialize readonly fields, Setup can not
- if  we need some Cleanup then having a Setup+Cleanup seems more intuitive than ctor+finalizer (BDN won't call Dispose if the class implements IDisposable)

when it comes to "putting tests into their own class" I think that it would simply create too many types and it would be harder to find the benchmarks in our reporting tool (in BenchView we have a namespace -> type name -> benchmark name tree)

Reply:
Then for consistency why not always use Setup?

Reply:
Honestly, I never thought about it from the consistency point of view. I think that it's better to keep it simple (don't use Setup if you don't need to).

Reply:
I think that we should discuss and agree on benchmarking best practices and guidelines and write them down into docs.